### PR TITLE
Remove custom type system

### PR DESCRIPTION
**Pull request by aroffringa**
See #1100 .

### CODE REVIEW

File: casa/Containers/RecordInterface.cc [hunk 1]
```diff
@@ -703,16 +702,9 @@ RecordNotice::RecordNotice (NoticeType changeType, uInt fieldNumber)
...
-    // This function returns the "Notice" type, retrieved
-    // from the "type registry".
-    return Register(this);
-}
-
-int RecordNotice::operator== (const Notice& that) const
-{
-    if (type() != that.type()) {
+    if (typeid(*this) != typeid(that)) {
 	return 0;
```
Comment (gervandiepen) [inline casa/Containers/RecordInterface.cc:708]: Maybe more logical to redturn False

File: measures/Measures/MeasIERS.cc [hunk 2]
```diff
@@ -41,6 +41,8 @@
 #include <casacore/casa/BasicSL/String.h>
 #include <casacore/casa/System/AppState.h>
 
+#include <atomic>
```
Comment (gervandiepen) [inline measures/Measures/MeasIERS.cc:44]: Why is this include necessary?

File: measures/Measures/MeasTable.cc [hunk 3]
```diff
@@ -53,6 +53,8 @@
 #include <casacore/casa/System/AipsrcValue.h>
 #include <casacore/casa/Utilities/Assert.h>
 
+#include <atomic>
```
Comment (gervandiepen) [inline measures/Measures/MeasTable.cc:56]: Why is this include necessary?

File: ms/MSOper/MSSummary.cc [hunk 4]
```diff
@@ -821,7 +821,7 @@ void MSSummary::listAntenna (LogIO& os, Bool verbose) const
         os << "z";
         os << endl;
         vector<MPosition> antPos = _msmd->getAntennaPositions();
-        Bool posIsITRF = antPos[0].type() != MPosition::ITRF;
+        Bool posIsITRF = antPos[0].getRefPtr()->getType() != MPosition::ITRF;
```
Comment (gervandiepen) [inline ms/MSOper/MSSummary.cc:824]: Why not antPos[0].getRef().getType() as done in MSMetaData.cc above?

File: ms/MSSel/MSSpWindowIndex.cc [hunk 5]
```diff
@@ -142,7 +142,7 @@ Vector<Int> MSSpWindowIndex::matchFreq(const Vector<MFrequency>& chanFreq,
     if (freqMatch(row)) {
       for (uInt chan=0; chan<nChan; chan++) {
 	freqMatch(row) = (freqMatch(row) &&
-			  chanFreq(chan).type() == rowChanFreq(chan).type() &&
+			  chanFreq(chan).getRefPtr()->getType() == rowChanFreq(chan).getRefPtr()->getType() &&
```
Comment (gervandiepen) [inline ms/MSSel/MSSpWindowIndex.cc:145]: Same remark as above; also for several files hereafter.

**Comment by gervandiepen** [comment]
In #1100 you comment that getRefPtr()->getType() returns an enum, but isn't it returning a uInt (from MRBase or MeasRef)?

File: measures/Measures/MeasIERS.cc (hunk 2 above)
Comment (aroffringa) [reply measures/Measures/MeasIERS.cc:44]: A std::atomic is used in line 100.

File: measures/Measures/MeasTable.cc (hunk 3 above)
Comment (aroffringa) [reply measures/Measures/MeasTable.cc:56]: Also because of use of std::atomic

File: ms/MSOper/MSSummary.cc (hunk 4 above)
Comment (aroffringa) [reply ms/MSOper/MSSummary.cc]: It's not clear to me when to use which one. I've changed it now.

**Comment by aroffringa** [comment]
> In #1100 you comment that getRefPtr()->getType() returns an enum, but isn't it returning a uInt (from MRBase or MeasRef)?

It's slightly confusing, it does return enum values like ITRF, but you're right that technically it returns an uInt. Not sure why.

**Review by gervandiepen — APPROVED** [review]

**Pull request merged**